### PR TITLE
Use `conda_package_streaming` for tarball extraction to avoid umask problems

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Should help with reports like https://github.com/conda/conda/issues/13953#issuecomment-2147948267 and https://github.com/conda-forge/miniforge/issues/506#issue-1968846490.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -69,6 +72,36 @@ def test_extract_conda_pkgs(tmp_path: Path):
...
+        return tar
+
+    tarbytes = empty_tarfile_bytes(name="naughty_umask", mode=0o777)
+    process = subprocess.Popen(
+        [CONDA_EXE, "constructor", "--prefix", tmp_path, "--extract-tarball"],
+        stdin=subprocess.PIPE,
+        umask=0o022,
+    )
+    process.communicate(tarbytes.getvalue())
+    rc = process.wait()
```
Comment (marcoesters) [inline tests/test_main.py:98]: Could `run_conda` be expanded to add stdin strings or is it not worth the effort?

File: tests/test_main.py [hunk 2]
```diff
@@ -69,6 +72,36 @@ def test_extract_conda_pkgs(tmp_path: Path):
...
+    assert rc == 0
+    if sys.platform != "win32":
+        mode = (tmp_path / "naughty_umask").stat().st_mode
+        assert not mode & stat.S_IWGRP, "%o" % mode
```
Comment (marcoesters) [inline tests/test_main.py:102]: This only checks group writing permissions. This is the origin of the bug, but shouldn't the entire umask be tested?

File: tests/test_main.py (hunk 1 above)
Comment (jaimergp) [reply tests/test_main.py:98]: If we need to send stdin in one more test, I'll do that :P

File: tests/test_main.py (hunk 2 above)
Comment (jaimergp) [reply tests/test_main.py:102]: Maybe. I copied the logic from Daniel's PR but maybe I can simply test the resulting chmod. Should be `0o777-0o22` right?

Comment (jaimergp) [reply tests/test_main.py]: New commit added, hopefully with the expected changes :)

**Review by marcoesters — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**